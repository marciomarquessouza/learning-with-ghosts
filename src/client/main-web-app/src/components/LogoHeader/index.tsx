import React from 'react'
import Link from 'next/link'
import Image from 'next/image'

import LogoImg from '../../../public/img/logo.png'

export interface LogoHeaderProps {
    url?: string
    altText?: string
}

const DEFAULTS = {
    url: '/',
    altText: 'Ghost Town - Learning with Ghosts',
}

function LogoHeader({ url = DEFAULTS.url, altText = DEFAULTS.altText }: LogoHeaderProps) {
    return (
        <Link href={url}>
            <Image src={LogoImg} alt={altText} width={150} height={170} />
        </Link>
    )
}

export default LogoHeader
