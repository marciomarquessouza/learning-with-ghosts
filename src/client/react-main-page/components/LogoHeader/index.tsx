import React from 'react'
import { Link } from 'react-router-dom'

export interface LogoHeaderProps {
    url?: string
    altText?: string
}

const DEFAULTS = {
    url: '/',
    altText: 'Ghost Town - Learning with Ghosts',
}

function LogoHeader({ url = DEFAULTS.url, altText = DEFAULTS.altText }: LogoHeaderProps) {
    return (
        <Link to={url}>
            <img src="/img/logo.png" alt={altText} />
        </Link>
    )
}

export default LogoHeader
