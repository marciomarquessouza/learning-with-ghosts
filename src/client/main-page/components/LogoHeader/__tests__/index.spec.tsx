import React from 'react'

import { render } from '@testing-library/react'

import LogoHeader from '..'

jest.mock('react-router-dom', () => ({
    ...(jest.requireActual('react-router-dom') as any),
    Link: ({ children }: { children: React.ReactNode }) => <a>{children}</a>,
}))

describe('components >> LogoHeader', () => {
    it('should render the LogoHeader component properly', () => {
        const { container } = render(<LogoHeader />)
        expect(container).toMatchSnapshot('LogoHeader-default')
    })
})
