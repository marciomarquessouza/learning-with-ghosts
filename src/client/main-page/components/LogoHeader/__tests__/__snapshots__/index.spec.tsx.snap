// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> LogoHeader should render the LogoHeader component properly: LogoHeader-default 1`] = `
<div>
  <a>
    <img
      alt="Ghost Town - Learning with Ghosts"
      src="/img/logo.png"
    />
  </a>
</div>
`;
